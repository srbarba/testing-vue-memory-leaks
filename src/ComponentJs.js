import { computedUserName } from './state/userState'

export default {
  props: {
    text: {
      type: String,
      default: 'Dummy component',
    },
  },
  computed: {
    userName: computedUserName,
  },
  render(h) {
    return h('v-card', [
      this.text,
      h(
        'div',
        this.$t('Hello {userName}', { userName: this.userName }).toString()
      ),
    ])
  },
}
