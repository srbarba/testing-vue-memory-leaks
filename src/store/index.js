export default {
  state: () => ({
    user: {
      name: undefined,
    },
  }),
  getters: {
    userName: (state) => state.user.name,
  },
  mutations: {
    setUser(state, user) {
      state.user = user
    },
  },
  actions: {},
}
