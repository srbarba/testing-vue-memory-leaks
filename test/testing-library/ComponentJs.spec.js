import { render } from '@testing-library/vue'
import ComponentJs from '../../src/ComponentJs'
import store from '../../src/store'

describe('ComponentJs', () => {
  it('renders correctly', () => {
    const wrapper = render(ComponentJs, { store })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
