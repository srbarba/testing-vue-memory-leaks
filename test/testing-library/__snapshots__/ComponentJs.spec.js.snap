// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentJs renders correctly 1`] = `"<div>Hello, undefined</div>"`;
